recipes/kfserving.rb

```diff
@@ -0,0 +1,208 @@
+  source "knative-serving-crds.yml.erb"
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+end
+
+template "/home/#{node['kube-hops']['user']}/knative/knative-serving.yaml" do
+  source "knative-serving.yml.erb"
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+end
+
+template "/home/#{node['kube-hops']['user']}/knative/knative-istio.yaml" do
+  source "knative-istio.yml.erb"
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+end
+
+kube_hops_kubectl 'apply-knative-serving-crds' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  url "/home/#{node['kube-hops']['user']}/knative/knative-serving-crds.yaml"
+end
+
+kube_hops_kubectl 'apply-knative-serving' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  url "/home/#{node['kube-hops']['user']}/knative/knative-serving.yaml"
+end
+
+kube_hops_kubectl 'apply-knative-istio-controller' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  url "/home/#{node['kube-hops']['user']}/knative/knative-istio.yaml"
+end
+
+# Cert-manager
+
+template "/home/#{node['kube-hops']['user']}/cert-manager.yaml" do
+  source "cert-manager.yml.erb"
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+end
+
+kube_hops_kubectl 'apply-cert-manager' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  flags "--validate=false"
+  url "/home/#{node['kube-hops']['user']}/cert-manager.yaml"
+end
+
+bash 'wait-for-cert-manager' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  environment ({ 'HOME' => ::Dir.home(node['kube-hops']['user']) })
+  code <<-EOH
+    kubectl wait --for=condition=Ready pods --all -n cert-manager --timeout=60s
+    sleep 15s
+    EOH
+end
+
+# KFServing
+
+template "/home/#{node['kube-hops']['user']}/kfserving.yaml" do
+  source "kfserving.yml.erb"
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+end
+
+kube_hops_kubectl 'apply-kfseving' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  url "/home/#{node['kube-hops']['user']}/kfserving.yaml"
+end
+
+# Hops system config
+
+hopsworks_endpoint = node['kube-hops']['pki']['ca_api'].split(':', 2)
+hopsworks_ip = hopsworks_endpoint.first
+hopsworks_port = hopsworks_endpoint.last
+
+template "/home/#{node['kube-hops']['user']}/hops-system.yaml" do
+  source "hops-system.yml.erb"
+  owner node['kube-hops']['user']
+  group node['kube-hops']['group']
+  variables({
+    'hopsworks_ip': hopsworks_ip,
+    'hopsworks_port': hopsworks_port
+  })
+end
+
+kube_hops_kubectl 'apply-hops-system' do
+  user node['kube-hops']['user']
+  group node['kube-hops']['group']
+  url "/home/#{node['kube-hops']['user']}/hops-system.yaml"
+end
+
+# Model Serving Admission Controller
+
+directory "/home/#{node['kube-hops']['user']}/model-serving-webhook" do
```

Comment:
Same `/srv/hops/` comment, actually not that I think about it, it would probably be great to group all the subcomponents in `/srv/hops/kfserving`